### ISSUE TITLE
RegExp.$1 not get the right value

### ISSUE THREAD

**Issue by Fndroid** [javascript-obfuscator/javascript-obfuscator#829]
When I tried to use RegExp by calling ``.test()``, I got different results when the code was obfuscatied. 

Here is the sample code to reproduce the issue: [gist](https://gist.github.com/Fndroid/8d509eecce73c9d1ee063115239fd194)

## Expected Behavior

The code above should log the host(``127.0.0.1:54637``) of the url to the console

## Current Behavior

By running multiple times of ``npm run test``, I got different results.
![image](https://user-images.githubusercontent.com/16091562/100983176-edb54580-3583-11eb-9178-47921fd565c4.png)

## Steps to Reproduce

1. clone the code from [gist](https://gist.github.com/Fndroid/8d509eecce73c9d1ee063115239fd194)
2. install deps ``npm install``
3. run ``npm run test`` 

## Your Environment

* Obfuscator version used: 2.9.4
* Node version used: v12.16.3

# Stack trace

no error throw in this case.

## Minimal working example that will help to reproduce issue

[gist](https://gist.github.com/Fndroid/8d509eecce73c9d1ee063115239fd194)

**Comment by da411d** [comment]
I did some test and there is what i get:
```
// Test 1
/(.+)/.test("12345");
console.log(RegExp.$1); //> 12345

// Test 2
/(.+)/.test("12345");
"a".replace(/=+$/, "");
console.log(RegExp.$1); //> 12345

// Test 3
/(.+)/.test("12345");
"a==".replace(/=+$/, "");
console.log(RegExp.$1); //> 
```

![image](https://user-images.githubusercontent.com/12462454/101209861-2d138d00-367d-11eb-957c-feda240597d3.png)
![image](https://user-images.githubusercontent.com/12462454/101209676-d1e19a80-367c-11eb-9656-eee1b058e8b0.png)


Looks like `RegExp.$1` resets after other regex. Such used in rc4 or base64 decode helper.
I think problem is in [AtobTemplate.ts](https://github.com/javascript-obfuscator/javascript-obfuscator/blob/808e7b2b302127f206694795caf01c2785b3f31d/src/custom-code-helpers/string-array/templates/string-array-calls-wrapper/AtobTemplate.ts#L11). Replacing with regex should be replaced with something else. Maybe substring

**Comment by sanex3339** [comment]
I'll try to remove padding characters from the `base64` encoded strings, and as result - remove this regexp that just trims these padding characters during decoding.

**Comment by sanex3339** [comment]
Released as `2.10.1`
